### PR TITLE
support charset input params and change default charset as utf8

### PR DESCRIPTION
**Pull request by ylwu-amzn**
### Description
Read more details https://github.com/opensearch-project/ml-commons/issues/1666
 
### Issues Resolved
Resolve https://github.com/opensearch-project/ml-commons/issues/1666
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Zhangxunmt — DISMISSED** [review]

**Comment by Zhangxunmt** [comment]
lgtm

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java [hunk 1]
```diff
@@ -435,17 +430,19 @@ public void testOpenAITextEmbeddingModel() throws IOException, InterruptedExcept
         responseMap = parseResponseToMap(response);
         taskId = (String) responseMap.get("task_id");
         waitForTask(taskId, MLTaskState.COMPLETED);
-        String predictInput = "{\n" + "  \"parameters\": {\n" + "      \"input\": \"The food was delicious\"\n" + "  }\n" + "}";
+        String predictInput = "{\n"
```
Comment (dhrubo-os) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:433]: Can we have another test with another charset? Maybe with ASCII or something else?

**Review by arjunkumargiri — COMMENTED** [review]
Nit: can we add more details into why this change is required.

**Comment by dhrubo-os** [comment]
> Nit: can we add more details into why this change is required.

I believe this is the [issue](https://github.com/opensearch-project/ml-commons/issues/1666)

**Review by dhrubo-os — APPROVED** [review]

**Review by Zhangxunmt — APPROVED** [review]

**Pull request merged**